packages/grafana-ui/src/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import { StatID } from '../../utils/statsCalculator';
+import { DisplayValue } from '../../types/index';
+
+export interface LegendBasicOptions {
+  isVisible: boolean;
+  asTable: boolean;
+}
+
+export interface LegendRenderOptions {
+  placement: LegendPlacement;
+  hideEmpty?: boolean;
+  hideZero?: boolean;
+}
+
+export type LegendPlacement = 'under' | 'right' | 'over'; // Over used by piechart
+
+export interface LegendOptions extends LegendBasicOptions, LegendRenderOptions {
+  stats?: StatID[];
```

Comment:
Can we keep stats out of the legend component?  It will need to be in the *panel*, but not necessary in the component.  Also StatID really just represents the StandardStatID(s) -- maybe we should rename it?

Reply:
just referencing https://github.com/grafana/grafana/pull/16468#discussion_r274495709 :)